worker/src/lib/managers/ReportManager.ts

```diff
@@ -0,0 +1,705 @@
+import { Env } from "../..";
+import { Result, err, ok } from "../util/results";
+import { Integration, ReportStore } from "../db/ReportStore";
+
+export class ReportManager {
+  private resendApiKey: Env["RESEND_API_KEY"];
+
+  constructor(private reportStore: ReportStore, private env: Env) {
+    this.resendApiKey = env.RESEND_API_KEY;
+  }
+
+  public async sendReports() {
+    const { data: allReports, error: allReportsErr } =
+      await this.reportStore.getReports();
+
+    if (allReportsErr) {
+      return err(`Error fetching reports: ${allReportsErr}`);
+    }
+
+    if (!allReports) {
+      return ok(`No reports found`);
+    }
+    const { error: sendEmailErr } = await this.sendReportEmails(allReports);
+    const { error: sendSlackErr } = await this.sendReportSlacks(allReports);
+
+    if (sendEmailErr) {
+      return err(`Failed to send report emails: ${sendEmailErr}`);
+    }
+
+    if (sendSlackErr) {
+      return err(`Failed to send report slacks: ${sendSlackErr}`);
+    }
+
+    return ok(null);
+  }
+
+  private async sendReportEmails(
+    reports: Integration[]
+  ): Promise<Result<null, string>> {
+    const promises = reports.map(async (report) => {
+      const settings = report.settings as {
+        emails: string[];
+        slack_channels: string[];
+      };
+      if (settings.emails.length > 0) {
+        const { error: emailResErr } = await this.sendReportEmail(
+          report,
+          settings.emails
+        );
+        if (emailResErr) {
+          console.error(`Error sending email: ${emailResErr}`);
+          throw new Error(emailResErr); // Throw the error to catch it outside
```

Comment:
this is not caught. Please just return the `err` type and change the result type to be `Result<string, string>`

Reply:
Oh, this was there in the alerts feature previously itself, so didn't take a look at it that well. My bad, will change it. Thank you!
Though, I have a question - shouldn't `Result<null, string>` be fine since SuccessResult is null and ErrorResult in the string?

https://github.com/Helicone/helicone/blob/44cb325051de67502ff114cd2a82fe9707f1980d/worker/src/lib/util/results.ts#L3-L12